### DISCUSSION TITLE
Change approach : make uBlock an application instead of a browser plugin ( a.k.a. : save uBlock from Manfest V3)

### DISCUSSION THREAD

**Discussion by thealio** [uBlockOrigin/uBlock-issues#3023] [Ideas] [answered]
Hi
I wanna be pretty short and go directly to the point of this discussion.

Referring to #2261 

Manifest V3 will limit the power and the functions of browsers extensions , especially adblockers.

The new uBlock Origin Manifest V3 will be actually less powerful and less usable .

Also , the browser-puglin nature of the project limits this to be used only  for chromium and firefox based browsers, kicking out the rest of the other browsers.

I' like to suggest another development approach: made Ublock a standalone application , that works in background, becoming browsers- independent.

This is the only solution to the problems I mentioned above in my opinion.

**Comment by MasterKia** [comment]
Are you willing to put the effort?

**Reply by thealio** [reply]
If you're talking about donations , of course.
Otherwise, I don't understand your point, because to develope a less featured extension will mean lots of efforts, because you 'll have to arrange it to manifest v3 limitations, and answer to a lot of users that will not understand why uBlock don't block ads like before, and all the related issues.

**Reply by gorhill** [reply]
> I' like to suggest another development approach: made Ublock a standalone application

@thealio Why ask someone else to implement your idea? If you strongly believe your idea is the way to go, why not just work on it?

**Reply by thealio** [reply]
Because you're the main Dev, and this is a discussion tread to share ideas.
ublock will be replaced by uBlock lite, that will not be as powerful as the manifest v2 version, and I think developing a bad copy of the manifest v2 version is not a solution.
Your answer is silly, the discussion aims to suggest and share idea, your "do it yourself" answer has no sense in my opinion

**Reply by gorhill** [reply, marked as answer]
> your "do it yourself" answer has no sense in my opinion

You idea is trivial, it has been talked over and over over the years.

If I am not doing this now, it's not because I could not have come up with such a trivial idea. It's because I do not have to the time and motivation to undertake such involved and difficult project.

I know how content blockers work, and I know the huge amount of work this would require, and the fact that this would require to deal with security-sensitive issues. I have no interest in getting involved in all this, I have a real life outside GitHub. AdGuard offers such product, you can already get this from them.

Whoever thinks it's a good idea, please undertake it yourself, good open source projects are not born out of a random person on the internet suggesting another person to implement _their_ idea, it's people scratching _their own_ itch in the first place.

People asking that I work on their idea has always been a recurring theme, hence why I use to put a warning against this in the [project's contributing file](https://github.com/gorhill/uBlock/blob/d75a1582ef3cff6644a7e87530ea9bda41af62e3/CONTRIBUTING.md#before-you-submit):

> Do NOT
> ...
> - Submit "revolutionary ideas".
> ...

TLDR: Please scratch your own itch, don't ask others to do it for you. uBO is GPLv3, whatever is found to be useful in it can be reused in another GPLv3 project.

**Reply by krystian3w** [reply]
I might only suggest that someone write a user manual for the uBo in the "Node" version:

- how to update the base filters in it
- how to change your subscription
- how to add a completely customised list like ~~letsblock.it~~
- how to save something to the My Filters panel as it exists there
  - whether firewall mode (medium/hard) can be used in it

Maybe then you* (* - 2-nd person plural) won't need to write a uBlock Origin Browser from scratch based on Electron.js or NW.js.

**Reply by thealio** [reply]
@gorhill 
I just suggested something that seemed to me a solution for a problem.
This is a discission thread where people can gently ask , to receive gently answers.
Nobody should blame you for agree or disagree to users ideas to improve the project.
I'm not asking you to do something I should do myself, this is intended only a brainstorming thread.
BTW , YOU are the main developer , so YOU have to "scratch your own itch"  on this project, not me, of course your can agree or disagree with users' ideas and reject them , no problem.
The problem sussist if you answer with such things like "do it by yourself" or "scratch your own itch", because I'm not a developer, you are, and this kind of answers sound like very silly and not helpful at all.

**Reply by Yuki2718** [reply]
> If you're talking about donations , of course.

Donation doesn't help anything. All of us are volunteers spending spare time for this project and are not working for $$$.

**Reply by stephenhawk8054** [reply]
> YOU have to "scratch your own itch"

The idea is suggested by you. This is not his own itch, and he doesn't have plans for it. The inside codes and security models between an extension and a standalone program are 2 different worlds.

This is a voluntary project. Everyone involved here are volunteers, contributions are done by each one's work including detailed reports and resolutions for the issues. This is not donation model where anyone can throw the money and ask others to work on some random ideas.

**Reply by garry-ut99** [reply]
>  Thealio: The problem sussist if you answer with such things like "do it by yourself" or "scratch your own itch", because I'm not a developer, you are, and this kind of answers sound like very silly and not helpful at all.

True. **gorhill** shouldn't immediately reply with attitude . Such discussions/proposals happen, there is no need go straight to accusation that someone didn't do something himself, this just poisons immediately the discussion. Just pointing out this is not possible or unfeasible is enough.

He should stop escalating and de-escalate instead. As a uBO creator here, there was no need for https://github.com/uBlockOrigin/uBlock-issues/discussions/3023#discussioncomment-7866804 and https://github.com/uBlockOrigin/uBlock-issues/discussions/3023#discussioncomment-7867379 in the first place, which is what started all these off-topic comments.

**Comment by krystian3w** [comment]
Plugins died in 2020 (Finally Flash death date), now is used only to DRM content or open PDF as long Google confirm use `pdf.js` instead own implementation as plugin.

**Reply by thealio** [reply]
I'm not talking about a plugin like flash.
I'm talking about a standalone applications

**Reply by krystian3w** [reply]
But uBo MV2 is not Plugin at all.

**Comment by Yuki2718** [comment]
Do you even understand what you are suggesting? If you think converting browser extension (based on javascript) to native app is so trivial, try to do by yourself first. Only then you will understand what you're suggesting.

**Comment by ghost** [comment]
MV3 api is limited, but outside the browser there is no api at all.

**Comment by krystian3w** [comment]
I doubt that turning uBo into a native application like AdGuard will be easy.
I tend to rule out a fork based on Electron.js or NW.js.

**Comment by Sp0kzz** [comment]
brave shields is built-in and uses uBo filters

**Reply by krystian3w** [reply]
But no support `:nth-child` / `:nth-of-type`…, so 30+ filters can not by used without rewrite to something else e.g. with `:has-text` (by human, not simple script in Rust).

**Comment by dm17** [comment]
Since https://github.com/uBlockOrigin/uBlock-issues/issues/3563 is locked, I wanted to write this in some relevant thread... Not because I'm asking for free engineering resources, but because I wanted to see what people think about it. If all of the uBo supported browsers have become significantly less private and/or secure than ungoogled-chromium, then doesn't that make ungoogled-chromium the best candidate for continued uBo support? We all know about the new Mozilla ToS. We don't yet have an unbraved Brave. And the consensus seems that uBo has better/more functionality than Brave Shields. And last time I checked my ungoogled-chromium browser significantly outperformed the rest on web.basemark.com. Considering the Mozilla ToS, I don't see how it could be deemed a safe long-term bet for continued uBo freedom. Is there something I'm missing here? 

BTW, I'm currently running ungoogled-chromium with the latest uBo without issue on any website so far.

Relevant info:
https://github.com/uBlockOrigin/uBlock-issues/discussions/2977#discussioncomment-9528691
https://github.com/ungoogled-software/ungoogled-chromium/issues/662

**Reply by Yuki2718** [reply]
Ungoogled or not, Chromium, not Chrome, drops the support for blocking webRequest or MV2.

**Reply by dm17** [reply]
Based on my last link there - I don't think we know that yet for ungoogled-chromium - do we?

**Reply by Yuki2718** [reply]
All Chromium-based browsers are affected, unless one explicitly tries to keep MV2, which requires significant effort to do.

**Reply by stephenhawk8054** [reply]
> Is there something I'm missing here?

uBO won't be converted to whatever program outside of the browsers. This has been declined many times.

We literally don't have any more free time for extra project here. If anyone wants to "continue uBo freedom", you can do it yourself. uBO is a GPLv3 FOSS project, including the extension and the lists, you can use whatever inside with your projects.

For example, go to here https://github.com/anfragment/zen and contribute whatever you can.

> what people think about it.

Many people think about the same idea, it's a trivial idea, but none wants to execute the idea themselves, just come here and ask the same again and again. You don't need to ask for uBO "permissions" to do anything, just start to do it, like the above project.

**Reply by dm17** [reply]
>uBO won't be converted to whatever program outside of the browsers. This has been declined many times.

Right, and if everyone accepts that, then aren't discussions here a perfect launchpad for many such offshoots - considering the interest & popularity in uBo? One could say the same for a Brave Sheilds fork for chromium - if such a thing is possible; I do not know.

**Reply by stephenhawk8054** [reply]
You can make any projects yourself, start to build things yourself, start to attract community yourself. uBO literally started like that, https://github.com/anfragment/zen started like that (did you check that project yet?). That project dev did not need to come here and ask "how about doing this/that". They just did it.

If you want to ask any technical details of uBO that you don't understand, you are free to ask here. Asking about starting a brand new project will just drown out other legitimate questions about uBO's technical and websites' issues, with the same answer again and again.
